src/main/java/SuperPeer.java

```diff
@@ -7,51 +6,88 @@
 import java.io.InputStreamReader;
 
 public class SuperPeer {
-
     //TODO: Add logger instead of system.out
     public static final String TAG = SuperPeer.class.getCanonicalName();
 
     private static final String EXIT_CMD = "exit";
     private static final String CLOSE_CHANNEL_CMD = "close";
 
-
     JavaMeshManager mm;
     private boolean isRunning = true;
     private TransactionsManager tm;
 
     public static void main(String[] args) {
-        SuperPeer p = new SuperPeer();
+        if (args.length > 0 && (args[0].equals("-h") || args[0].equals("--headless"))) {
+            // Doesn't read from STDIN if run with `-h` or `--headless`.
+            // Useful for backgrounding.
+            SuperPeer interactiveMode = new SuperPeer(false);
```

Comment:
are these variable names reversed?

Reply:
I believe so. Comments might be as well. Sorry about that.

Reply:
Fixed.